src/lib.rs

```diff
@@ -1,57 +1,177 @@
 //! # reedline `\|/`
-//! A readline replacement written in Rust
+//! # A readline replacement written in Rust
 //!
-//! ## Example (Simple REPL)
+//! ## Basic example
 //!
 //! ```rust,no_run
-//! use reedline::{Reedline, DefaultPrompt, Signal};
+//! // Create a default reedline object to handle user input
 //!
-//! let mut line_editor = Reedline::new();
-//! let prompt = DefaultPrompt::default();
+//! use reedline::{DefaultPrompt, Reedline, Signal};
 //!
-//! loop {
-//!     let sig = line_editor.read_line(&prompt).unwrap();
-//!     match sig {
-//!         Signal::CtrlD | Signal::CtrlC => {
-//!             line_editor.print_crlf().unwrap();
-//!             break;
-//!         }
-//!         Signal::Success(buffer) => {
-//!             // process `buffer`
-//!             println!("We processed: {}", buffer);
-//!         }
-//!         Signal::CtrlL => {
-//!             line_editor.clear_screen().unwrap();
+//! fn main() {
+//!     let mut line_editor = Reedline::new();
+//!     let prompt = DefaultPrompt::default();
+//!
+//!     loop {
+//!         let sig = line_editor.read_line(&prompt).unwrap();
+//!         match sig {
+//!             Signal::Success(buffer) => {
+//!                 println!("We processed: {}", buffer);
+//!             }
+//!             Signal::CtrlD | Signal::CtrlC => {
+//!                 line_editor.print_crlf().unwrap();
+//!                 break;
+//!             }
+//!             Signal::CtrlL => {
+//!                 line_editor.clear_screen().unwrap();
+//!             }
 //!         }
 //!     }
 //! }
+//! ```
+//! ## Integrate with custom Keybindings
+//!
+//! ```rust,no_run
```

Comment:
Some tests might actually not hit the terminal and don't run in an endless I/O loop like the core example. So we might carefully consider at some point switching to live tests checking at least the constructors.

Reply:
Okay